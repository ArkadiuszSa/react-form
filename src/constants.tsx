class Constants {
  public API_BASE;
  constructor() {
    this.API_BASE = "https://event-list-server.herokuapp.com/api";
  }
}

export default new Constants();
