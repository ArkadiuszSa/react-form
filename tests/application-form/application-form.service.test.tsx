import Service from "../../src/modules/user/pages/application-form/application-form.service";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import constants from "../../src/constants";
import * as moment from "moment";

const apiBase = constants.API_BASE;

var mock = new MockAdapter(axios);
mock.onGet(apiBase + "/happening/id").reply(200, {
  title: "Happening Name",
  days: ["2018-11-11"]
});

const applicationFormService = new Service();

describe("ApplicationForm-service", () => {
  test("should fetch happening and return transformed data", done => {
    applicationFormService.getHappening("id").then(res => {
      expect(res).toMatchSnapshot();
      done();
    });
  });

  test("should return axios post promise", done => {
    applicationFormService
      .AddNewApplication("application")
      .then(() => {
        done();
      })
      .catch(() => {
        done();
      });
  });

  test("should positively validate form and return boolean result", () => {
    const application = {
      firstName: "John",
      lastName: "Doo",
      email: "johndo@gmail.com",
      date: "2018-11-18",
      avaibleDates: [moment("2018-11-18")]
    };
    const result = applicationFormService.validateForm(application);
    expect(result.isError).toBe(false);
    expect(result.errors).toMatchSnapshot();
  });

  test("should negatively validate form and return boolean result with errors", () => {
    const application = {
      firstName: "",
      lastName: "",
      email: "",
      date: "",
      avaibleDates: [moment("2018-11-18")]
    };
    const result = applicationFormService.validateForm(application);
    expect(result.isError).toMatchSnapshot();
    expect(result.errors).toMatchSnapshot();
  });
});
