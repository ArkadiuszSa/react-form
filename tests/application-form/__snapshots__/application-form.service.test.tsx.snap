// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationForm-service should fetch happening and return transformed data 1`] = `
Object {
  "avaibleDates": Array [
    "2018-11-10T23:00:00.000Z",
  ],
  "date": "2018-11-11",
  "selectedDate": "2018-11-10T23:00:00.000Z",
  "title": "Sign up for Happening Name",
}
`;

exports[`ApplicationForm-service should negatively validate form and return boolean result with errors 1`] = `true`;

exports[`ApplicationForm-service should negatively validate form and return boolean result with errors 2`] = `
Object {
  "dateErr": "Requires date",
  "emailErr": "Requires valid email",
  "firstNameErr": "Requires first name",
  "lastNameErr": "Requires last name",
}
`;

exports[`ApplicationForm-service should positively validate form and return boolean result 1`] = `
Object {
  "dateErr": "",
  "emailErr": "",
  "firstNameErr": "",
  "lastNameErr": "",
}
`;
